lib/spack/llnl/util/tty/log.py

```diff
@@ -31,12 +37,26 @@
 control = re.compile('(\x11\n|\x13\n)')
 
 
+@contextmanager
+def background_safe():
+    signal.signal(signal.SIGTTOU, signal.SIG_IGN)
+    yield
+    signal.signal(signal.SIGTTOU, signal.SIG_DFL)
+
+
+def _is_background_tty():
+    """Return True iff this process is backgrounded and stdout is a tty"""
+    if sys.stdout.isatty():
+        return os.getpgrp() != os.tcgetpgrp(sys.stdout.fileno())
+    return False  # not writing to tty, not background
+
+
 def _strip(line):
     """Strip color and control characters from a line."""
     return _escape.sub('', line)
 
 
-class keyboard_input(object):
+class _keyboard_input(object):
```

Comment:
This is intended to be usable outside here.  If `termios` is available, it allows us to turn off line-buffered input and respond to `v`.  If it is not available, this *still* marks a region where we expect keyboard input, and the user can just type `v` - `RETURN`.  It's not intended to be private.